packages/core/stories/floating-platform/custom-floating-ui-platform.stories.tsx

```diff
@@ -0,0 +1,257 @@
+        (
+          {
+            style,
+            open,
+            top,
+            left,
+            width,
+            height,
+            position,
+            ...rest
+          }: RootComponentProps,
+          ref: ForwardedRef<HTMLElement>
+        ) => {
+          const FloatingRoot = (
+            /* In thise case to avoid Flash of Unstyled Text (FOUT) in the tooltip, due to being in an iframe, we are always rendering the tooltip.
+             * We are visually hiding it until it is open to 'eagerly load' the font
+             * In a more realistic example e.g. desktop you would take a different approach e.g. re-using an existing window which could have fonts loaded in advance
+             * Alternatively you could use the Font Loader API to check fonts have been loaded before showing the tooltip, or use a better fallback system font which would cause less layout shift
+             */
+            <FloatingComponentWindow
+              style={{
+                ...style,
+                top,
+                left,
+                border: "none",
+                padding: 0,
+                position: "fixed",
+                width,
+                height,
+                ...(!open ? offscreenStyles : undefined),
+              }}
+            >
+              {/* max-content allows tooltip to size itself, the TooltipWindow will resize to fit this */}
+              <div
+                style={{ width: "max-content" }}
+                {...rest}
+                ref={ref as Ref<HTMLIFrameElement>}
+              />
+            </FloatingComponentWindow>
+          );
+
+          // In this case tooltip is portalled back to the root document this may not be the case if tooltips were opened as new windows
+          return rootBody ? createPortal(FloatingRoot, rootBody) : null;
+        }
+      ),
+    [rootBody]
+  );
+
+  return (
+    <NewWindow ref={setIframe} style={{ height: 300 }}>
+      <div style={{ padding: 10 }}>
+        <StackLayout gap={3}>
+          <H3>This is an iframe with a button</H3>
+          <Text>It represents a portalled window within an application</Text>
+          {showExtraContent && <H3>Some Extra Content!</H3>}
+          <FloatingPlatformProvider platform={customPlatform} animationFrame>
+            <FloatingComponentProvider Component={FloatingUIComponent}>
+              <Tooltip {...props} open>
+                <Button
+                  onClick={() => {
+                    setShowExtraContent((old) => !old);
+                  }}
+                >
+                  Click to show extra content
+                </Button>
+              </Tooltip>
+              <ComboBoxNext disabled={false} source={source} />
+              <DropdownNext source={source} />
+            </FloatingComponentProvider>
+          </FloatingPlatformProvider>
+        </StackLayout>
+      </div>
+    </NewWindow>
+  );
+};
+
+export const CustomFloatingUiPlatform: Story<TooltipProps> = (
```

Comment:
This example flashes for a moment before disappearing when viewing it preview storybook link

Reply:
Very strange. This is a problem on Firefox but not chrome. All the html for the story disappears without any error in the console.

This is quite a contrived example, trying to find a way to represent multiple windows via iframes and protalling in a browser so I'm not expecting consumers to do the same thing. If it's a just an issue for this story on Firefox then I don't think we need to worry too much

Reply:
Not a blocker in that case